core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -72,7 +72,7 @@ class ReplicaFetcherThread(name: String,
       metrics,
       time,
       "replica-fetcher",
-      Map("broker-id" -> sourceBroker.id.toString).asJava,
+      Map("thread-id" -> name).asJava,
```

Comment:
thread-id always include the broker-id, but broker-id itself is not sufficient in distinguishing metrics from the threads to the same broker.